spec/integration/recipes/lwrp_inline_resources_spec.rb

```diff
@@ -45,6 +45,33 @@ def action_a
     end
   end
 
+  context "with an inline resource with a property that shadows the enclosing provider's property" do
+    class LwrpShadowedPropertyTest < Chef::Resource::LWRPBase
+      PATH = ::File.join(Dir.tmpdir, "shadow-property.txt")
+      use_automatic_resource_name
+      actions :fiddle
+      property :content
+      action :fiddle do
+        file PATH do
+          content new_resource.content
+          action [:create, :delete]
+        end
+      end
+    end
+
+    after { File.delete(LwrpShadowedPropertyTest::PATH) rescue nil }
```

Comment:
I wanted to make sure the file got deleted even if something went wrong with the `file` resource's `:delete` action. That will mostly never happen, so the FileNotFound exception case is normal.

As I type that, I like it less, when I can just do `if File.exists?`.